### ISSUE TITLE
Enable the possibility to include "preferences" to Users using UI and API 

### ISSUE THREAD

**Issue by susangz** [TheHive-Project/TheHive#1097]
### Request Type
Feature Request

### Work Environment
TheHive Training VM - Beta Version

### Problem Description
I would like to add preferences to the users in order to use them in a Responder to perform the more suitable actions. I've seen that attribute is already included in the User model but it doesn't appear in the UI and it is not allowed through the API REST. 

### Steps to Reproduce
1. Create a user using the API:
curl -XPOST -H 'Authorization: Bearer Vi3WqXISouEXsgkxzMXQIXmnFHaGSVhM' -H 'Content-Type: application/json' http://localhost:9000/api/user -d '{"login": "testuser", "name":"User created using API","roles":["read","write"],"password":"testuser","preference":"{\"Group\":\"GroupA\"}"}'
{"tableName":"user","type":"AttributeCheckingError","errors":[{"name":"user.preference","type":"UnknownAttributeError","message":"Unknown attribute user.preference: {\"type\":\"JsonInputValue\",\"value\":\"{\\\"Group\\\":\\\"GroupA\\\"}\"}","value":{"type":"JsonInputValue","value":"{\"Group\":\"GroupA\"}"}}]}

**Comment by veeral-patel** [comment]
@susangz Hey - what is your use case for this?

**Comment by susangz** [comment]
I need to enforce an incident management workflow with users belonging to different groups:
-  users in groupA will introduce information about a detected cyber incident  
- once they have finished their task (it would be also great if a task could be assigned to a group of users), users in groupB will classify the incident impact/severity based on the information provided by groupA
- once classified the incident, users in groupC will supervise the classification and will give the OK to proceed with the reporting of the incident
- if groupC gives the OK for an incident, users of groupD will prepare the report with the information of that incident and will notify it to Competent Authorities

I'm thinking to implemente it using webhooks (to know when a user has completed a task and to monitor user activity) and through Responders, controlling which actions are invoked so they are performed only if the user has the right permissions.

The problem is that, if I'm not wrong, in TheHive there is no user groups or roles management. Consequently, I need to have this information about the group where the user belongs to in some place. I've thought user "preferencies" could be an option since it is already included in the User model, but I'm open to any other idea/suggestion...  ;-)

Thanks!!

**Comment by veeral-patel** [comment]
@susangz Would something like solve your problem?

1. Being able to define custom statuses. In your case, you'd have DETECTED, CLASSIFIED, CLASSIFICATION_APPROVED, WRITING_REPORT, REPORT_COMPLETED

2. Whenever a group finishes their work, they change the status and assign the case to the lead from the next group. The lead for the next group may make previous case members read-only.

> I'm thinking to implemente it using webhooks (to know when a user has completed a task and to monitor user activity) and through Responders, controlling which actions are invoked so they are performed only if the user has the right permissions.

Sorry I don't see how responders are involved here?

**Comment by susangz** [comment]
Hi! Thanks for the suggestions, I haven't  thought in those options. I guess when you say custom statuses you mean using Tags in the case, right? Probably we will need to manage the users permissions from our component based just on the usernames.

About the responders is e.g. to send the case to some external tool such as Alfresco to prepare and manage reports once all the information about the incident has been collected and analysed.

**Comment by veeral-patel** [comment]
@susangz 

> I guess when you say custom statuses you mean using Tags in the case, right?

Nope, I mean instead of just having Open and Closed statuses, let the user add additional statuses.
